web/Services/AzureStorage.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace BricksAndHearts.Services
+{
+    public interface IAzureStorage
+    {
+        Task CreateContainerAsync(string varType, int id);
+        Task DeleteContainerAsync(string varType, int id);
+        Task UploadFileAsync(IFormFile file, string varType, int id);
+        Task<List<string>> ListFilesAsync(string varType, int id);
+        Task<(Stream data, string fileType)> DownloadFileAsync(string varType, int id, string blobFilename);
+        Task DeleteFileAsync(string varType, int id, string blobFilename);
+    }
+
+    public class AzureStorage : IAzureStorage
+    {
+        private readonly string _storageConnectionString;
+        private readonly string _baseContainerName;
+        private readonly ILogger<AzureStorage> _logger;
+
+        public AzureStorage(IConfiguration configuration, ILogger<AzureStorage> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("BlobConnectionString");
+            _baseContainerName = configuration.GetValue<string>("BlobContainerName");
+            _logger = logger;
+        }
+
+        private string GetContainerName(string varType, int id)
+        {
+            return _baseContainerName + varType + id.ToString();
+        }
+
+        private async Task<BlobContainerClient> GetContainerClient(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobContainerClient containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            if (!containerClient.Exists())

```

Comment:
I wonder if we should have a separate `GetOrCreateContainerClient` method with this logic in, just to be clearer what's going on?

Reply:
Having double-checked the code, the only time CreateContainer is actually used now is within GetContainerClient (containers are created the first time they are accessed); I'll combine the two methods into a single GetOrCreateContainerClient, which should be clearer and avoid repeating so many steps.